src/Storages/StorageMySQL.cpp

```diff
@@ -95,8 +95,12 @@ Pipe StorageMySQL::read(
         sample_block.insert({ column_data.type, column_data.name });
     }
 
+    if (!context_->getSettingsRef().external_storage_max_read_rows)
```

Comment:
```suggestion
    /// Check if setting is enabled, otherwise use common `max_block_size` setting.
    if (!context_->getSettingsRef().external_storage_max_read_rows)
```
A comment needed. Also it is the only place where it is checked. But default value of setting is now `0`.  Then in all other places it works incorrectly. Here it almost does not make any difference to use `settings.max_block_size` instead of `max_block_size` which is passed into `read` method. It means that you can move this check into constructor of StreamSettings.

Reply:
I think the settings should be consistent. 

The judgment here is to keep old users supporting the old configuration.

If upgrade to this version config the new settings is the right usage.